### ISSUE TITLE
on win7, the settings are uploaded and syned, however, on ubuntu, the extensions are not synced!

### ISSUE THREAD

**Issue by jonygli** [shanalikhan/code-settings-sync#213]
**Visual Studio Code Version :** [ 1.9.1]
**Code Settings Sync Version :** [ 2.4.3]
**Operating System :** [ ubuntu 16.04 ]
**Occurs On:** [Download]
**Proxy Enabled:** [ Yes  ]
**GIst Id:** [ 4f15b34b7eed738e4854cb6f37409b73  ]

on win7, the settings are uploaded and syned, however, on ubuntu, the extensions are not synced!

**Comment by jonygli** [comment]
it seems the extensions can be synced and installed only at the 1st time. after then, extensions can't be synced and installed.

**Comment by shanalikhan** [comment]
is there any exception logged in console.

**Comment by xubo-bj** [comment]
I met the same problem, upload from **win7** then download to **ubuntu 16.04**,  _another extensions_ "**Syncing**" can work

**Comment by UserRandom** [comment]
Same issue, although between different operating systems. Started relatively recently. Possibly related to #206 .

**Comment by generalov** [comment]
Probably it could be solved if change the `lastDownload` option value in User settings from the `false` to the same string date value as `lastUpload` has (or may be just to the empty string). 

I've got the same ussue on Ubuntu 16.10 and debugged the extension. 
The exception is raised on https://github.com/shanalikhan/code-settings-sync/blob/master/src/extension.ts#L303
```
let lastDownloadStr: string = syncSetting.lastDownload.toString();
```
 because the syncSetting.lastDownload has the `undefined` value.

**Comment by shanalikhan** [comment]
@generalov Thanks!
i have replaced it by additional undefined check.
```
let lastUploadStr: string = (syncSetting.lastUpload) ? syncSetting.lastUpload.toString(): "";
let lastDownloadStr: string = (syncSetting.lastDownload) ? syncSetting.lastDownload.toString(): "";
```
#206 also tried to solve similar work, i will publish this in newer version so you all can check it out.

**Comment by shanalikhan** [comment]
I have release a update for Code Insiders, if anyone is using Insiders please let me know if it worked i will release for public, I have tested and its working for me.

**Comment by shanalikhan** [comment]
Public build released v2.4.4 Fixing this issue
Let me know if there is some problem.

**Comment by generalov** [comment]
Thanks!
